digitalocean/resource_digitalocean_monitor_alert_test.go

```diff
@@ -0,0 +1,284 @@
+	region = "fra1"
+	size   = "s-1vcpu-1gb"
+	tags    = [digitalocean_tag.test.name]
+  }
+
+  resource "digitalocean_monitor_alert" "%s" {
+	alerts  {
+	  email 	= ["benny@digitalocean.com"]
+	}
+	window      = "%s"
+	type        = "%s"
+	compare     = "GreaterThan"
+	value       = 95
+	tags        = [digitalocean_tag.test.name]
+	description = "%s"
+}
+`
+
+	testAccAlertPolicyAddDroplet = `
+resource "digitalocean_droplet" "web" {
+	image  = "ubuntu-20-04-x64"
+	name   = "web-1"
+	region = "fra1"
+	size   = "s-1vcpu-1gb"
+  }
+
+resource "digitalocean_droplet" "web2" {
+	image  = "ubuntu-20-04-x64"
+	name   = "web-2"
+	region = "fra1"
+	size   = "s-1vcpu-1gb"
+}
+
+
+resource "digitalocean_monitor_alert" "%s" {
+	alerts  {
+	  email 	= ["benny@digitalocean.com"]
+      %s
+	}
+	window      = "%s"
+	type        = "%s"
+	compare     = "GreaterThan"
+	value       = 95
+	entities    = [digitalocean_droplet.web.id, digitalocean_droplet.web2.id]
+	description = "%s"
+  }
+`
+)
+
+func TestAccDigitalOceanMonitorAlert(t *testing.T) {
+	var randName = randomTestName()
+	resourceName := fmt.Sprintf("digitalocean_monitor_alert.%s", randName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                  func() { testAccPreCheck(t) },
+		ProviderFactories:         testAccProviderFactories,
+		CheckDestroy:              testAccCheckDigitalOceanMonitorAlertDestroy,
+		PreventPostDestroyRefresh: true,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccAlertPolicy, randName, "", "5m", "v1/insights/droplet/cpu", "Alert about CPU usage"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "type", "v1/insights/droplet/cpu"),
+					resource.TestCheckResourceAttr(resourceName, "compare", "GreaterThan"),
+					resource.TestCheckResourceAttr(resourceName, "alerts.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "alerts.0.email.0", "benny@digitalocean.com"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDigitalOceanMonitorAlertEmailAlerts(t *testing.T) {
+	var randName = randomTestName()
+	resourceName := fmt.Sprintf("digitalocean_monitor_alert.%s", randName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                  func() { testAccPreCheck(t) },
+		ProviderFactories:         testAccProviderFactories,
+		CheckDestroy:              testAccCheckDigitalOceanMonitorAlertDestroy,
+		PreventPostDestroyRefresh: true,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccAlertPolicy, randName, " ", "5m", "v1/insights/droplet/cpu", "Alert about CPU usage"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "type", "v1/insights/droplet/cpu"),
+					resource.TestCheckResourceAttr(resourceName, "compare", "GreaterThan"),
+					resource.TestCheckResourceAttr(resourceName, "alerts.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "alerts.0.email.0", "benny@digitalocean.com"),
+				),
+			},
+		},
+	})
+}
```

Comment:
These tests look identical. Is one supposed to test something different than the other?

Reply:
Looks like something that should be deleted. Please do so. Thanks for noticing! @andrewsomething helped add and fix tests so I think it's good.

Reply:
You can create an alert without Slack and Email (which was done with Support alerts using only tags (48ca9ce).). I don't really know what I had in mind with the duplicate test here.